document/FileFormat.mdk

```diff
@@ -2408,23 +2412,132 @@ This allows tools to include information about the stack frame that is not expli
+
+### `threadId` property { #annotatedCodeLocation-threadId }
+
+An `annotatedCodeLocation` object may contain a property named `threadId` whose value is an integer which
+identifies that thread that was executing when the execution of a code flow reached the location specified by this object.
+If this `annotatedCodeLocation` does not occur within a `codeFlow`, the `threadId` property shall be absent.
+
 ### `message` property { #annotatedCodeLocation-message }
 
 An `annotatedCodeLocation` object may contain a property named `message` whose value is a string that describes
 the significance of this location within a particular context.
 
+### `kind` property { #annotatedCodeLocation-kind }
+
+An `annotatedCodeLocation` object may contain a property named `kind` whose value is a string that provides
+a descriptive identifier that categorizes the annotation.
+
+If present, the `kind` property shall have one of the following values, with the specified meanings:
+
+* `"alias"`: This location defines an additional name for a variable defined in a declaration.
+
+* `"branch": At this location, a branch in the execution path occurred.
+
+* `"call"`: This location is the site of a function or method call.
+
+* `"callReturn"`: This location is the target of a return from a function or method.
+
+* `"declaration"`: The location declares a variable.
+
+* `"functionEnter"`: This location is an entry point to a function or method.
+
+* `"functionExit": This location is an exit point from a function or method (for example, a return statement).
+
+* `"sanitizer"`: At this location,  data that entered the system from outside (for example, from user input)
+is ensured to be safe.
+
+* `"sink"`: At this location, data leaves the system (for example, by being written to the console or sent over a network).
+
+* `"source"`: At this location, data enters the system (for example, by being provided by a user or read from a file on disk).
+
+* `"usage"`: At location, data is used. 
+
 ~ IsoExample
 Suppose an analysis tool produces a result which states that a piece of data from an insecure source
 has been used at a particular location.
 The tool might provide a "related location" (&sect;[#result-relatedLocations]) whose value is
 an `annotatedCodeLocation` object with the message "Insecure data entered the system here". 
 ~
 
+### `essential` property { #annotatedCodeLocation-essential }
+
+An `annotatedCodeLocation` object may contain a property named `essential` whose value is a Boolean value
+that specifies whether this `annotatedCodeLocation` is essential to understanding the `codeFlow` object
+(&sect;[#codeFlow]) in which it occurs.
+If this `annotatedCodeLocation` does not occur within a `codeFlow`, the `essential` property shall be absent.
+
+~ IsoNote
+A viewer might use this property to offer the user an abbreviated view of a lengthy code flow,
+displaying only the most important locations in the code flow.
```

Comment:
why didn't you say 'only the essential locations in the code flow'? this is as precise/descriptive as what you have and emphasizes the property name (which is something you tend to do elsewhere). Just curious.